### PR TITLE
Update APIs to conform to Swift 3 API Design Guidelines

### PR DESCRIPTION
**Pull request by jpsim**
This _should_ be exhaustive, but it's hard to say without a straightforward way to verify whether or not this satisfies all the suggestions in Apple's design guidelines.

Can you please review @norio-nomura ?

### CODE REVIEW

File: Source/SourceKittenFramework/Request.swift [hunk 1]
```diff
@@ -117,83 +116,85 @@ private var sourceKitWaitingRestoredSemaphore = DispatchSemaphore(value: 0)
...
+        return nil
     }
-    return nil
 }
 
 /// Represents a SourceKit request.
 public enum Request {
     /// An `editor.open` request for the given File.
-    case EditorOpen(File)
+    case EditorOpen(file: File)
```
Comment (norio-nomura) [inline Source/SourceKittenFramework/Request.swift]: Do we continue using uppercase member here?

Comment (jpsim) [inline Source/SourceKittenFramework/Request.swift]: you're right, these should probably be lowercased. will do.

**Comment by norio-nomura** [comment]
I have checked public and changed APIs, but not yet on unchanged or unexposed APIs.

**Comment by jpsim** [comment]
Still reviewing this @norio-nomura?

**Comment by norio-nomura** [comment]
Oh, sorry, I got sidetracked to how can we replace `SourceKitRepresentable` with another type.
This is looks good to me. :+1:

**Comment by jpsim** [comment]
Cool, will merge this then and re-target #267 towards `master` so we can move on to that.

**Pull request merged**